modules/service/src/test/scala/lucuma/odb/graphql/feature/perScienceObservationCalibrations.scala

```diff
@@ -0,0 +1,386 @@
+      _           <- recalculateCalibrations(pid, when)
+      obs1Before  <- queryObservation(oid1)
+      obs2Before  <- queryObservation(oid2)
+      group1Id    =  obs1Before.groupId.get
+      // Delete obs1, keep obs2, add obs3
+      _           <- setObservationInactive(oid1)
+      oid3        <- createFlamingos2LongSlitObservationAs(pi, pid, List(tid3))
+      _           <- recalculateCalibrations(pid, when)
+      group1Exists <- queryGroupExists(group1Id)
+      obs2After   <- queryObservation(oid2)
+      obs3After   <- queryObservation(oid3)
+    } yield {
+      assert(!group1Exists)
+      assert(obs2After.groupId.isDefined)
+      assert(obs3After.groupId.isDefined)
+    }
+
+  test("New F2 observation creates group even with inactive F2 observation present"):
+    for {
+      pid         <- createProgramAs(pi)
+      tid1        <- createTargetWithProfileAs(pi, pid)
+      tid2        <- createTargetWithProfileAs(pi, pid)
+      oid1        <- createFlamingos2LongSlitObservationAs(pi, pid, List(tid1))
+      _           <- recalculateCalibrations(pid, when)
+      obs1Before  <- queryObservation(oid1)
+      group1Id    =  obs1Before.groupId.get
+      // Set observation1 inactive
+      _           <- setObservationInactive(oid1)
+      // Create new F2 observation
+      oid2        <- createFlamingos2LongSlitObservationAs(pi, pid, List(tid2))
+      _           <- recalculateCalibrations(pid, when)
+      group1Exists <- queryGroupExists(group1Id)
+      obs2After   <- queryObservation(oid2)
+    } yield {
+      assert(!group1Exists)
+      assert(obs1Before.groupId =!= obs2After.groupId)
+      assert(obs2After.groupId != Some(group1Id))
+    }
+
+  test("Recalculation handles GMOS observation correctly"):
+    for {
+      pid         <- createProgramAs(pi)
+      tid         <- createTargetWithProfileAs(pi, pid)
+      oid         <- createGmosNorthLongSlitObservationAs(pi, pid, List(tid))
+      _           <- recalculateCalibrations(pid, when)
+      obs         <- queryObservation(oid)
+    } yield {
+      assertEquals(obs.groupId, None)
+    }
+
+  test("Observation takes telluric group's position when group is deleted"):
+    for {
+      pid  <- createProgramAs(pi)
+      tid  <- createTargetWithProfileAs(pi, pid)
+      oid  <- createFlamingos2LongSlitObservationAs(pi, pid, List(tid))
+      _    <- recalculateCalibrations(pid, when)
+      obs  <- queryObservation(oid)
+      telluricGroupId = obs.groupId.get
+      telluricGroup <- queryGroup(telluricGroupId)
+      // Change to GMOS
+      _    <- updateObservationMode(oid, "gmosNorthLongSlit")
+      _    <- recalculateCalibrations(pid, when)
+      obsAfter <- queryObservation(oid)
+    } yield {
+      assertEquals(obsAfter.groupId, telluricGroup.parentId)
+      assertEquals(obsAfter.groupIndex, Some(telluricGroup.parentIndex))
+    }
+
+  test("Observation preserves parent group position when nested telluric group is deleted"):
```

Comment:
I see these tests for checking that an observation takes the telluric group's place in the parent group when the telluric group is deleted, but is there a test checking that the telluric group takes the observation's place when the telluric group is created and the observation placed into it?

Reply:
Cool. LGTM